### PR TITLE
feat: Add WaiterTypedError compliance to UnknownAWSHttpServiceError

### PR DESCRIPTION
**Pull request by jbelkins**
## Issue \#
This PR is in conjunction with waiters work, and will merge to a new feature branch: `feat/waiters`.  The waiter feature branch should only merge when the Waiters feature (by the same branch name) does in `smithy-swift`.

## Description of changes
- `UnknownAWSHttpServiceError` now captures error code on creation, and complies with the `WaiterTypedError` protocol defined in `smithy-swift`.
- Two codegen tests have been adjusted for `WaiterTypedError`-related changes.

## New/existing dependencies impact assessment, if applicable
No new dependencies were added to this change.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AWSClientRuntime/Sources/Errors/UnknownAWSHttpServiceError.swift [hunk 1]
```diff
@@ -7,6 +7,8 @@ import ClientRuntime
 
 /// AWS specific Service Error structure used when exact error could not be deduced from the `HttpResponse`
 public struct UnknownAWSHttpServiceError: AWSHttpServiceError, Equatable {
+    public var _errorCode: String?
```
Comment (epau) [inline AWSClientRuntime/Sources/Errors/UnknownAWSHttpServiceError.swift:10]: Can we add docs for error code here?

File: AWSClientRuntime/Sources/Errors/UnknownAWSHttpServiceError.swift [hunk 2]
```diff
@@ -23,10 +25,21 @@ public struct UnknownAWSHttpServiceError: AWSHttpServiceError, Equatable {
 }
 
 extension UnknownAWSHttpServiceError {
-    public init(httpResponse: HttpResponse, message: String? = nil, requestID: String? = nil) {
+    public init(
+        httpResponse: HttpResponse,
+        message: String? = nil,
+        requestID: String? = nil,
+        errorCode: String? = nil
+    ) {
```
Comment (epau) [inline AWSClientRuntime/Sources/Errors/UnknownAWSHttpServiceError.swift:33]: non-blocking, but would you mind adding a doc comment for this initializer while you're here?

File: AWSClientRuntime/Sources/Errors/UnknownAWSHttpServiceError.swift [hunk 3]
```diff
@@ -23,10 +25,21 @@ public struct UnknownAWSHttpServiceError: AWSHttpServiceError, Equatable {
...
         self._statusCode = httpResponse.statusCode
         self._headers = httpResponse.headers
         self._requestID = requestID ?? httpResponse.headers.value(for: X_AMZN_REQUEST_ID_HEADER)
         self._message = message
     }
 }
+
+extension UnknownAWSHttpServiceError: CodedError {
+
+    public var errorCode: String? { _errorCode }
```
Comment (epau) [inline AWSClientRuntime/Sources/Errors/UnknownAWSHttpServiceError.swift:44]: Can we add docs for error code here?

**Review by epau — APPROVED** [review]
Looks good, just ask for doc comments before merging

File: AWSClientRuntime/Sources/Errors/UnknownAWSHttpServiceError.swift (hunk 1 above)
Comment (jbelkins) [reply AWSClientRuntime/Sources/Errors/UnknownAWSHttpServiceError.swift]: Added

File: AWSClientRuntime/Sources/Errors/UnknownAWSHttpServiceError.swift (hunk 2 above)
Comment (jbelkins) [reply AWSClientRuntime/Sources/Errors/UnknownAWSHttpServiceError.swift]: Added

File: AWSClientRuntime/Sources/Errors/UnknownAWSHttpServiceError.swift (hunk 3 above)
Comment (jbelkins) [reply AWSClientRuntime/Sources/Errors/UnknownAWSHttpServiceError.swift]: Added

**Pull request merged**